packages/teams-js/src/public/pages.ts

```diff
@@ -1096,4 +1103,117 @@ export namespace pages {
         : false;
     }
   }
+
+  /**
+   * Provides APIs to interact with the button in hosts
+   *
+   * @beta
+   */
+  export namespace responseButton {
+    /**
+     * The required information that the app needs to let host apps know.
+     *
+     * @beta
+     */
+    export interface ResponseInfo {
+      /**
+       * The unique ID helps the host and apps identify which button it refers to.
+       *
+       * @beta
+       */
+      responseId: string;
+      /**
+       * Id of File on OneDrive
+       *
+       * @beta
+       */
+      oneDriveFileId: string;
+      /**
+       * The exact message where the action was invoked from
+       *
+       * @beta
+       */
+      messageId: string;
+
+      /**
+       * The conversation where the action was invoked from
+       *
+       * @beta
+       */
+      conversionId: string;
+    }
+
+    /**
+     * Notify host apps to show the response button.
+     * @param params - Parameters sent to host apps contain the required information.
+     *
+     * @beta
+     */
+    export function showResponseButton(params: ResponseInfo): Promise<void> {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!params) {
```

Comment:
would also be good to make sure sure each param is not null, along with any other validation that we can do (no empty strings, etc.)